### PR TITLE
[test] [swift] URL/Message EndpointListener tests

### PR DESCRIPTION
**Pull request by jayahariv**
* URL/Message Endpoint + collection tests
* Make fixes

### CODE REVIEW

File: Swift/ReplicatorConfiguration.swift [hunk 1]
```diff
@@ -406,17 +422,23 @@ public struct ReplicatorConfiguration {
...
+        if legacyConstructor {
+            c = CBLReplicatorConfiguration(database: self.database._impl, target: target.toImpl())
+            c.channels = channels
+            c.documentIDs = documentIDs
+            c.pushFilter = self.filter(push: true)
+            c.pullFilter = self.filter(push: false)
+            
+        } else {
+            c = CBLReplicatorConfiguration(target: target.toImpl())
+        }
```
Comment (pasin) [inline Swift/ReplicatorConfiguration.swift:435]: I think only do this will work as well.
```
c = CBLReplicatorConfiguration(target: target.toImpl())
```

File: Swift/ReplicatorConfiguration.swift [hunk 2]
```diff
@@ -476,5 +498,11 @@ public struct ReplicatorConfiguration {
     }
     
     var collectionConfigs = [Collection: CollectionConfiguration]()
+    
+    // [collection.fullName : collection]
+    var collectionMap = [String: Collection]()
```
Comment (pasin) [inline Swift/ReplicatorConfiguration.swift:503]: I forgot the rule of Swift's dictionary key. Is collectionConfigs alone not working?

Comment (jayahariv) [reply Swift/ReplicatorConfiguration.swift:503]: we don't have unique collection objects for the given name/scopeName. Every time a new object will be created so comparing with object might not work.

Comment (pasin) [reply Swift/ReplicatorConfiguration.swift:503]: I think a proper way to implement this is to have Collection class implements Hashable.

Comment (jayahariv) [reply Swift/ReplicatorConfiguration.swift:503]: ok.

Comment (jayahariv) [reply Swift/ReplicatorConfiguration.swift:503]: Something is not right, even though collection was hashable, last time it showed some error. Which is not happening right now. Let me remove this and try again.

File: Swift/Tests/URLEndpointListenerTest.swift [hunk 3]
```diff
@@ -791,7 +791,7 @@ class URLEndpointListenerTest_Main: URLEndpointListenerTest {
         try stopListener()
     }
     
-    func testEmptyNetworkInterface() throws {
+    func _testEmptyNetworkInterface() throws {
```
Comment (pasin) [inline Swift/Tests/URLEndpointListenerTest.swift:794]: This seems to be unrelated to this PR. Is this by intention?

File: Swift/ReplicatorConfiguration.swift [hunk 4]
```diff
@@ -406,17 +414,15 @@ public struct ReplicatorConfiguration {
     
     func toImpl() -> CBLReplicatorConfiguration {
         let target = self.target as! IEndpoint
-        let c = CBLReplicatorConfiguration(database: self.database._impl, target: target.toImpl())
+        var c: CBLReplicatorConfiguration!
+        
+        c = CBLReplicatorConfiguration(target: target.toImpl())
```
Comment (pasin) [inline Swift/ReplicatorConfiguration.swift:419]: Do it in one line `var c = CBLReplicatorConfiguration(target: target.toImpl())`?

File: Swift/Collection.swift [hunk 5]
```diff
@@ -299,11 +299,18 @@ public final class Collection : CollectionChangeObservable, Indexable, Equatable
...
     }
     
     var isValid: Bool { _impl.isValid }
     
     let _impl: CBLCollection
     let _db: Database
     let _scope: Scope
+    
+    // The collection name prefixed with scope name. e.g., "scopeA.collectionA"
+    let fullName: String
```
Comment (pasin) [inline Swift/Collection.swift:312]: Is this still needed?

File: Swift/Collection.swift [hunk 6]
```diff
@@ -299,11 +299,18 @@ public final class Collection : CollectionChangeObservable, Indexable, Equatable
...
+    
+    // The default collection name prefixed with default scope name. "_default._default"
+    static let defaultCollectionFullName: String = "\(kCBLDefaultScopeName).\(kCBLDefaultCollectionName)"
```
Comment (pasin) [inline Swift/Collection.swift:315]: Is this still neeeded?

File: Swift/ReplicatorConfiguration.swift [hunk 7]
```diff
@@ -384,13 +384,21 @@ public struct ReplicatorConfiguration {
...
     
-    var defaultCollectionConfig: CollectionConfiguration {
+    var defaultCollectionConfig: CollectionConfiguration? {
         guard let col = try? self.database.defaultCollection() else {
             fatalError("Default collection is missing!")
         }
         
-        guard let colConfig = self.collectionConfigs[col] else {
-            fatalError("default collection config missing")
+        if let config = self.collectionConfigs[col] {
```
Comment (pasin) [inline Swift/ReplicatorConfiguration.swift:392]: I don't see the implementation that makes Collection a Hashable. Is it not required?

**Review by pasin — CHANGES_REQUESTED** [review]

File: Swift/Tests/URLEndpointListenerTest.swift (hunk 3 above)
Comment (jayahariv) [reply Swift/Tests/URLEndpointListenerTest.swift:794]: accidental

File: Swift/Collection.swift (hunk 5 above)
Comment (jayahariv) [reply Swift/Collection.swift:312]: No

File: Swift/Collection.swift (hunk 6 above)
Comment (jayahariv) [reply Swift/Collection.swift:315]: No,. will remove.

File: Swift/ReplicatorConfiguration.swift (hunk 7 above)
Comment (jayahariv) [reply Swift/ReplicatorConfiguration.swift:392]: it is already hashable. 
https://github.com/couchbase/couchbase-lite-ios/blob/master/Swift/Collection.swift#L286-L294

**Review by pasin — APPROVED** [review]
